bootloader/grub.go

```diff
@@ -457,3 +460,24 @@ func staticCommandLineForGrubAssetEdition(asset string, edition uint) string {
 	}
 	return string(cmdline)
 }
+
+// TrustedAssetsChain returns the list of relative paths to files inside
+// the bootloader's rootdir that are measured in the boot process in the
+// order of loading during the boot.
+func (g *grub) TrustedAssetsChain() ([]string, error) {
+	if !g.native {
+		return nil, fmt.Errorf("internal error: trusted assets called without native hierarchy")
+	}
+	if g.recovery {
+		return []string{
+			// recovery mode shim EFI binary
+			"EFI/boot/bootx64.efi",
+			// recovery mode grub EFI binary
+			"EFI/boot/grubx64.efi",
+		}, nil
+	}
+	return []string{
+		// run mode grub EFI binary
+		"EFI/boot/grubx64.efi",
```

Comment:
is this correct? In run mode, don't we boot first the recovery-mode shim, then the recovery-mode grub, which then chain-loads to the run mode grub? Or do we chain load in such a way that measurements are reset when the run mode grub is executed ?

Reply:
yes, you are right, though the current code in gadget/install is also confusing because we aren't measuring binaries themselves yet, just keys so it matters less. But it will matter very much if we measure binaries themselves

https://github.com/snapcore/snapd/blob/master/gadget/install/install.go#L166

Reply:
right, but this function should still return the full chain, even if we are only measuring keys from those binaries, right? Whether the policy is setup to measure the full binary or the key is immaterial to this code, the caller of this will use these assets as inputs to secboot along with the decision of whether to measure the binary or key IIRC.

Reply:
Perhaps I should adjust the name to TrustedAssets to avoid the confusion. Otherwise, returning both `bootx64.efi` and `grubx64.efi` for system-boot bootloader is equally confusing.

Reply:
Yes, "Chain" is a confusing. I think  if one reads the doc comment the behavior presented here is correct. That's why I +1ed the PR but I suppose I wasn't assuming this is meant to be the end of it. It means will need a different method to produce the chains that sealing consumes but that's ok I suppose.

Reply:
Per discussion, the name has been change to `TrustedAssets`.